### PR TITLE
Support saving deferred bindings with pivot data.

### LINKED ISSUE #1409: Pivot data support in deferred binding
Is there any way to create relation manager which has some pivot data and works with deffered binding?

Are there any chances for October to support this anytime soon?

### PR DESCRIPTION
**Pull request by SergeyKasyanov**
This PR gives the ability to save many-to-many relations with pivot data using deferred bindings.

Fixes https://github.com/octobercms/october/issues/1409

### CODE REVIEW

**Comment by SergeyKasyanov** [comment]
related to https://github.com/octobercms/october/pull/5466

**Comment by mjauvin** [comment]
Is morphToMany taken into account with this?

File: src/Database/Traits/DeferredBinding.php [hunk 1]
```diff
@@ -146,7 +147,11 @@ protected function commitDeferredOfType($sessionKey, $include = null, $exclude =
             $relationObj = $this->$relationName();
 
             if ($binding->is_bind) {
-                $relationObj->add($slaveModel);
+                if ($relationType === 'belongsToMany') {
```
Comment (mjauvin) [inline src/Database/Traits/DeferredBinding.php]:
```suggestion
                if (in_array($relationType, ['belongsToMany', 'morphToMany'])) {
```

Comment (SergeyKasyanov) [reply src/Database/Traits/DeferredBinding.php]: May be there must be 'morphedByMany' too?

**Comment by SergeyKasyanov** [comment]
Added morphToMany and morphedByMany. But now test for deferred unbinding for morphToMany is not passing.

File: src/Database/Migrations/2021_01_19_000001_Db_Add_Pivot_Data_To_Deferred_Bindings.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+use October\Rain\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class DbAddPivotDataToDeferredBindings extends Migration
+{
+
+    public function up()
+    {
+        Schema::table('deferred_bindings', function (Blueprint $table) {
+            $table->text('pivot_data')->nullable()->after('slave_id');
```
Comment (LukeTowers) [inline src/Database/Migrations/2021_01_19_000001_Db_Add_Pivot_Data_To_Deferred_Bindings.php]:
```suggestion
            $table->mediumText('pivot_data')->nullable()->after('slave_id');
```

**Comment by SergeyKasyanov** [comment]
Now all tests passing fine.

**Comment by daftspunk** [comment]
Looking good, thanks for this!

**Pull request merged**